projects/core/src/lib/layer-collections/layer-collection.service.ts

```diff
@@ -45,4 +76,186 @@ export class LayerCollectionService {
             .getSessionStream()
             .pipe(mergeMap((session) => this.backend.getWorkflowMetadata(workflowId, session.sessionToken)));
     }
+
+    /**
+     * Add all layers (directly) contained in a layer collection to the current project.
+     */
+    addCollectionLayersToProject(collection: LayerCollectionDict): void {
+        const layersObservable = collection.items
+            .filter((layer) => layer.type === 'layer')
+            .map((layer) => layer as LayerCollectionLayerDict)
+            .map((layer) => this.resolveLayer(layer.id));
+
+        // TODO: lookup in parallel
+        zip(layersObservable)
+            .pipe(map((layers) => this.projectService.addLayers(layers)))
+            .subscribe();
+    }
+
+    /**
+     * Add a layer to the current project.
+     */
+    addLayerToProject(layerId: ProviderLayerIdDict): void {
```

Comment:
Maybe use `subscribeAndProvide` to make it optional to await on it?

We should move the method from he ProjectService to a general util then, I guess.

Reply:
8379c66470c4beb3fdc9118ef3a17eba367299f8